packages/react-router/lib/dom/ssr/components.tsx

```diff
@@ -784,24 +784,37 @@ import(${JSON.stringify(manifest.entry.module)});`;
 
   return isHydrated ? null : (
     <>
+      {manifest.sri ? (
+        <script
+          type="importmap"
+          dangerouslySetInnerHTML={{
+            __html: JSON.stringify({
+              integrity: manifest.sri,
+            }),
+          }}
+        />
+      ) : null}
       {!enableFogOfWar ? (
         <link
           rel="modulepreload"
           href={manifest.url}
           crossOrigin={props.crossOrigin}
+          integrity={manifest.sri?.[manifest.url]}
```

Comment:
I'm curious if we _need_ these if we already have all the assets represented in the importmap above?  It feels like it's just a bit more bloat to the HTML doc?

Reply:
I assume that any of our preload links and stuff from `<Link prefetch>` and our route module loading during navigations will fall back to using the importmap for integrity checks?

Reply:
We do need them, but they fallback to the map after the document has loaded.